license_manager/apps/subscriptions/management/commands/validate_num_catalog_queries.py

```diff
@@ -30,23 +30,37 @@ def handle(self, *args, **options):
         ]
 
         # Batch the catalog UUIDs and aggregate API response data in a set
-        distinct_catalog_query_ids = set()
+        all_response_data = {}
         for catalog_uuid_batch in chunks(distinct_catalog_uuids, constants.VALIDATE_NUM_CATALOG_QUERIES_BATCH_SIZE):
             response = EnterpriseCatalogApiClient().get_distinct_catalog_queries(catalog_uuid_batch)
-            distinct_catalog_query_ids.update(response['catalog_query_ids'])
-        distinct_catalog_query_ids = list(distinct_catalog_query_ids)
+            query_ids = response['catalog_query_ids']
+            for key in query_ids.keys():
+                if key in all_response_data.keys():
+                    all_response_data[key] += query_ids[key]
+                else:
+                    all_response_data[key] = query_ids[key]
+        distinct_catalog_query_ids = all_response_data.keys()
```

Comment:
```suggestion
           catalog_uuids_by_query_id = response['catalog_query_ids']
           for query_id, catalog_uuids in query_ids.items():
                all_response_data[query_id].extend(catalog_uuids)
```
Also consider renaming `all_response_data` to `all_catalog_uuids_by_query_id` and making it a `defaultdict(list)`.